std-bits/table/src/main/java/org/enso/table/aggregations/AggregateColumn.java

```diff
@@ -0,0 +1,96 @@
+package org.enso.table.aggregations;
+
+import org.enso.table.data.table.problems.AggregatedProblems;
+import org.enso.table.data.table.problems.Problem;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/***
+ * Interface used to define aggregate columns.
+ */
+public abstract class AggregateColumn {
+  private final String name;
+  private final int type;
+  private AggregatedProblems problems;
+
+  protected AggregateColumn(String name, int type) {
+    this.name = name;
+    this.type = type;
+    this.problems = null;
+  }
+
+  /***
+   * @return Name of the new column.
+   */
+  public final String getName() {
+    return name;
+  }
+
+  /***
+   * @return The type of the new column.
+   */
+  public int getType() {
+    return type;
+  }
+
+  public AggregatedProblems getProblems() {
+    return problems;
+  }
+
+  /***
+   * Compute the value for a set of rows
+   * @param rows - indexes to the rows in the source table to aggregate on
+   * @return aggregated value
+   */
+  public Object aggregate(int[] rows) {
+    return this.aggregate(Arrays.stream(rows).boxed().collect(Collectors.toList()));
+  }
+
+  /***
+   * Compute the value for a set of rows
+   * @param rows - indexes to the rows in the source table to aggregate on
+   * @return aggregated value
+   */
+  public abstract Object aggregate(List<Integer> rows);
+
+  protected void addProblem(Problem problem) {
+    if (problems == null) {
+      problems = new AggregatedProblems();
+    }
+    problems.add(problem);
+  }
+
+  protected static Long CastToLong(Object value) {
+    if (value instanceof Long) {
+      return (Long)value;
+    } else if (value instanceof Integer) {
+      return ((Integer)value).longValue();
+    } else if (value instanceof Byte) {
+      return ((Byte)value).longValue();
+    } else if (value instanceof Float && ((Float)value) % 1 == 0) {
```

Comment:
What's the purpose of the `((Float)value) % 1 == 0` check? Is it checking if it is an integer value? I'd suggest adding some temporary variable to give this a name to make the intent clearer.

Reply:
Still think a comment here would be good just to make the intention clear.